src/types.c

```diff
@@ -522,48 +567,66 @@ netplan_netdef_get_embedded_switch_mode(const NetplanNetDefinition* netdef)
     return netdef->embedded_switch_mode;
 }
 
-NETPLAN_INTERNAL gboolean
-netplan_netdef_get_delay_virtual_functions_rebind(const NetplanNetDefinition* netdef)
+NETPLAN_INTERNAL ssize_t
+_netplan_netdef_get_embedded_switch_mode(const NetplanNetDefinition* netdef, char* out_buffer, size_t out_buf_size)
 {
     g_assert(netdef);
-    return netdef->sriov_delay_virtual_functions_rebind;
+    return netplan_copy_string(netdef->embedded_switch_mode, out_buffer, out_buf_size);
 }
 
 NETPLAN_INTERNAL gboolean
-netplan_netdef_has_match(const NetplanNetDefinition* netdef)
+netplan_netdef_get_delay_virtual_functions_rebind(const NetplanNetDefinition* netdef)
 {
-    return netdef->has_match;
+    g_assert(netdef);
+    return netdef->sriov_delay_virtual_functions_rebind;
 }
 
-NETPLAN_INTERNAL NetplanNetDefinition*
-_netplan_netdef_get_sriov_link(const NetplanNetDefinition* netdef)
+__attribute((alias("netplan_netdef_get_delay_virtual_functions_rebind"))) NETPLAN_INTERNAL gboolean
+_netplan_netdef_get_delay_vf_rebind(const NetplanNetDefinition* netdef);
+
+NETPLAN_INTERNAL guint
+_netplan_netdef_get_vf_count(const NetplanNetDefinition* netdef)
```

Comment:
I'm sorry for the back and forth.. but I feel like we should actually drop this function, as we already have `_netplan_state_get_vrf_count_for_def`, which is more complete implementation wise, making use of the full "netplan_state" context (i.e. implicit sriov_link definitions).

Reply:
I don't have a strong opinion on that, so I'll drop it. We can add it later if we miss it...